### PR TITLE
KAFKA-8461: Wait for follower to join the ISR in testUncleanLeaderElectionDisabled Test

### PR DESCRIPTION
**Pull request by omkreddy**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala [hunk 1]
```diff
@@ -252,8 +252,17 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
 
     produceMessage(servers, topic, "third")
     waitUntilMetadataIsPropagated(servers, topic, partitionId)
-    servers.filter(server => server.config.brokerId == leaderId).map(server => shutdownServer(server))
+    //make sure follower server joins the ISR
+    TestUtils.waitUntilTrue(() => {
+      val partitionInfoOpt = followerServer.metadataCache.getPartitionInfo(topic, partitionId)
+      if (partitionInfoOpt.isDefined) {
```
Comment (abbccdda) [inline core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala]: Could we just do
`partitionInfoOpt.isDefined && partitionInfoOpt.get.basePartitionState.isr.size == 2` for the boolean condition here?

Comment (omkreddy) [reply core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala]: @abbccdda Thanks for the review. Updated the PR.

**Review by abbccdda — APPROVED** [review]
@omkreddy Thank you for the fix, LGTM

**Comment by abbccdda** [comment]
@ijuma could you also take a look? The test fails another time today:
https://builds.apache.org/job/kafka-pr-jdk11-scala2.12/5343/console

File: core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala [hunk 2]
```diff
@@ -252,8 +252,13 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
 
     produceMessage(servers, topic, "third")
     waitUntilMetadataIsPropagated(servers, topic, partitionId)
-    servers.filter(server => server.config.brokerId == leaderId).map(server => shutdownServer(server))
+    //make sure follower server joins the ISR
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala:254]: Is `waitUntilMetadataIsPropagated` needed above?

File: core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala [hunk 3]
```diff
@@ -252,8 +252,13 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
...
+    TestUtils.waitUntilTrue(() => {
+      val partitionInfoOpt = followerServer.metadataCache.getPartitionInfo(topic, partitionId)
+      partitionInfoOpt.isDefined && partitionInfoOpt.get.basePartitionState.isr.size == 2
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala]: nit: maybe we could be explicit and require that the ISR contains `followerId`?

File: core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala [hunk 4]
```diff
@@ -252,8 +252,13 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
...
+    }, "Inconsistent metadata after first server startup")
 
+    servers.filter(server => server.config.brokerId == leaderId).map(server => shutdownServer(server))
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala]: nit: `foreach` instead of `map`

**Review by hachikuji — APPROVED** [review]
LGTM. Left a few minor comments. Feel free to merge once resolved.

**Comment by omkreddy** [comment]
@hachikuji Thanks for the review. Test failures not related. merging to master and 2.3

**Pull request closed without merging**